### PR TITLE
Regtest network

### PR DESCRIPTION
**Pull request by alexeykoren**
Use regtest network for test
+ fmt and cleanup

### CODE REVIEW

File: src/commands/test.rs [hunk 1]
```diff
@@ -95,14 +99,29 @@ fn pause() {
...
 
-fn prepare_test(config: &Reader<AppConfig>, wallet: &mut Wallet, rpc_client: &mut ReqwestRpcClient) -> TxId {
+fn prepare_test(
+    config: &Reader<AppConfig>,
+    wallet: &mut Wallet,
+    rpc_client: &mut ReqwestRpcClient,
+) -> TxId {
     wallet.reset();
     sync_from_height(config.chain.nu5_activation_height, wallet, rpc_client);
+    mine_block(rpc_client, vec![], true); // mine Nu5 activation block
```
Comment (PaulLaux) [inline src/commands/test.rs:109]: consider comments above the line for consistency.

File: src/components/rpc_client.rs [hunk 2]
```diff
@@ -24,22 +24,21 @@ pub trait RpcClient {
...
-        hash: String,
+    /// The hash of the requested block in hex.
+    hash: String,
 
-        /// The number of confirmations of this block in the best chain,
-        /// or -1 if it is not in the best chain.
-        confirmations: i64,
+    /// The number of confirmations of this block in the best chain,
+    /// or -1 if it is not in the best chain.
+    confirmations: i64,
```
Comment (PaulLaux) [inline src/components/rpc_client.rs:32]: consider `confirmations: Option<u64>` and `None` to mark "not in the best chain"

File: src/components/rpc_client/reqwest.rs [hunk 3]
```diff
@@ -81,8 +100,9 @@ impl RpcClient for ReqwestRpcClient {
         let mut params: Vec<ParamType> = Vec::new();
         params.push(ParamType::String(hex::encode(txid.as_ref())));
         params.push(ParamType::Number(0)); // Verbosity
-        // params.push(ParamType::String(hex::encode(block_id.0.as_ref())));
-        let tx_hex: String = self.request(&RpcRequest::new_with_params("getrawtransaction", params))?;
+                                           // params.push(ParamType::String(hex::encode(block_id.0.as_ref())));
```
Comment (PaulLaux) [inline src/components/rpc_client/reqwest.rs:103]: do we need this comment?

File: src/components/transactions.rs [hunk 4]
```diff
@@ -34,33 +40,55 @@ pub fn mine_block(rpc_client: &mut dyn RpcClient, txs: Vec<Transaction>) -> (u32
...
 
-    let mut tx = Builder::new(TEST_NETWORK, /*wallet.last_block_height().unwrap()*/ BlockHeight::from_u32(1_842_421), wallet.orchard_anchor());
+    let mut tx = Builder::new(
+        REGTEST_NETWORK,
+        wallet.last_block_height().unwrap(),
+        wallet.orchard_anchor(),
+    );
 
-    let coinbase_value = 500000000;
+    let coinbase_value = 625000000;
```
Comment (PaulLaux) [inline src/components/transactions.rs:70]: consider `625_000_000` for consistency

File: src/components/transactions.rs [hunk 5]
```diff
@@ -146,9 +207,18 @@ pub fn create_transfer_tx(sender: Address, recipient: Address, amount: u64, wall
...
+pub fn template_into_proposal(
+    block_template: BlockTemplate,
+    mut txs: Vec<Transaction>,
+    activate: bool,
+) -> BlockProposal {
+    let coinbase = Transaction::read(
+        hex::decode(block_template.coinbase_txn.data)
+            .unwrap()
+            .as_slice(),
+        zcash_primitives::consensus::BranchId::Nu5,
```
Comment (PaulLaux) [inline src/components/transactions.rs:219]: This code is relevant for `nu5` but probably not for `nu6`
Consider naming the function `_nu5` or make it as a parameter.

File: src/components/transactions.rs [hunk 6]
```diff
@@ -157,17 +227,25 @@ pub fn template_into_proposal(block_template: BlockTemplate, mut txs: Vec<Transa
...
-            bytes
-        } else {
-            AUTH_COMMITMENT_PLACEHOLDER
-        }
-    }).collect::<AuthDataRoot>();
+    let auth_data_root = txs_with_coinbase
+        .iter()
+        .map(|tx| {
+            if tx.version().has_orchard() {
+                let bytes: [u8; 32] =
```
Comment (PaulLaux) [inline src/components/transactions.rs:241]: `: [u8;32]` is redundant

**Review by PaulLaux — COMMENTED** [review]
Very nice, added some minor comments.
+ Need to add `cargo clippy`, `cargo fmt` and `cargo test` (something basic) to github actions.

File: src/components/rpc_client.rs (hunk 2 above)
Comment (alexeykoren) [reply src/components/rpc_client.rs:32]: This part is copied from Zebra RPC, I think it is a good idea to keep it untouched for consistency and future updates

File: src/components/transactions.rs (hunk 5 above)
Comment (alexeykoren) [reply src/components/transactions.rs:219]: Good point

File: regtest-config.toml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+miner_address = 'tmLTZegcJN5zaufWQBARHkvqC62mTumm3jR'
+
+[network]
+network = "Regtest"
+
+# This section may be omitted when testing only Canopy
+[network.testnet_parameters.activation_heights]
+# Configured activation heights must be greater than or equal to 1,
+# block height 0 is reserved for the Genesis network upgrade in Zebra
+NU5 = 1
```
[the remark this answers is not in the document]
Comment (alexeykoren) [reply regtest-config.toml:11]: For Orchard/ZSA tests - not really. We only support Nu5+ plus very partially transaparent, so this is enough

**Review by PaulLaux — APPROVED** [review]

**Pull request merged**